src/clientHandler.ts

```diff
@@ -279,6 +279,11 @@ export class ClientHandler {
     return clients.get(this.clientName(document.toString()));
   }
 
+  public clientSupportsCommand(cmdName: string, document?: vscode.Uri): boolean {
+    const commands = this.getClient(document).client.initializeResult.capabilities.executeCommandProvider?.commands;
```

Comment:
In each of these `.client` calls, there is a possible exception where the extension is activated, the editor is activated, but the client is not ready yet because the language client and LS haven't initialized yet.

If we use the `onReady` method and chain to `.then()` and await the result we can be reasonably sure we won't throw exceptions.

This will be an infrequent case because our LS starts so quickly, but I encountered this when developing the module view. If a user has an active window/panel open that triggers this before the LS has started and is ready, then an error is shown to the client that they don't know what to do about.

Reply:
raised a separate PR with patch for this https://github.com/hashicorp/vscode-terraform/pull/778 then I can rebase this one, hopefully retaining the same ClientHandler API 🤞🏻

Reply:
Rebased this PR - PTAL